src/resolve_addr.rs

```diff
@@ -0,0 +1,165 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
+
+use futures::Async;
+use futures::Future;
+use futures::Poll;
+use std::error::Error;
+use std::fmt;
+use std::net::SocketAddr;
+use std::net::ToSocketAddrs;
+
+/// Go-style network address parsing. Returns a future.
+/// Examples:
+/// "192.0.2.1:25"
+/// ":80"
+/// "[2001:db8::1]:80"
+/// "198.51.100.1:80"
+/// "deno.land:443"
+pub fn resolve_addr(address: &str) -> ResolveAddrFuture {
```

Comment:
How do you decide whether to resolve ipv4/ipv6 or either ?

Reply:
Rust's resolution scheme to_socket_addrs() decides that. This procedure I've made is a bit hacky and just uses the first ip returned.